### ISSUE TITLE
LUA script doesn't return evaljs content on Ubuntu 16.04.

### ISSUE THREAD

**Issue by laranicolas** [scrapinghub/splash#901]
I am trying to scrap the HTML and the Window object from site using Splash V.3.3.1. Locally on OSX environment it works perfectly but when I deploy to Ubuntu 16.04 production server LUA scripts return the object only with 'html' key.

```
def parse(self, response):
    print(response.data)
```
Result:

```{'html': 'ALL HTML'}```

Script:
```
    def start_requests(self):
        """Starting place for request."""
        script = """
        function main(splash, args)
          assert(splash:go(args.url))
          assert(splash:wait(0.5))
          local data = splash:evaljs("window.typeOfObject")
          return {
            js = data,
            html = splash:html()
          }
        end
        """
        yield SplashRequest(self.url, self.parse, endpoint='execute', args={
            'wait': 1,
            'proxy': 'http://localhost:24000',
            'lua_source': script
        })
```

**Comment by Gallaecio** [comment]
Are the versions of Scrapy and Splash the same on both machines? Could you try on a newer version of Ubuntu?

**Comment by laranicolas** [comment]
Yes, they are same versions.

```
Scrapy==1.6.0
scrapy-splash==0.7.2
```

Making an upgrade now of production server is not an option.

**Comment by Gallaecio** [comment]
Are you using the same Splash instance in both cases?

**Comment by laranicolas** [comment]
Sorry what do you means with Splash instance?

**Comment by Gallaecio** [comment]
I mean the server or service providing Splash. Are you using the same `SPLASH_URL` setting on both cases, or are you using a different `SPLASH_URL` (pointing to a different Splash instance)?

**Comment by laranicolas** [comment]
Yes I am using the same `SPLASH_URL` locally and in production. The weird thing is that if I remove `evaljs` on production, means I don't user JS variables everything works fine.

**Comment by laranicolas** [comment]
Still having issues, this is what I see on log file while running:

```Bad request to Splash: {'info': {'message': 'Lua error: [string "..."]:3: network99', 'line_number': 3, 'error': 'network99', 'source': '[string "..."]', 'type': 'LUA_ERROR'}, 'description': 'Error happened while executing Lua script', 'error': 400, 'type': 'ScriptError'}```

Any idea?

**Comment by laranicolas** [comment]
When looking more specific under window js object it raises undefined error:

```2019-05-23 19:10:30 [scrapy_splash.middleware] WARNING: Bad request to Splash: {'info': {'line_number': 5, 'error': 'JS error: "TypeError: undefined is not an object (evaluating \'window. typeOfObject.data.overview\')"', 'message': '[string "..."]:5: JS error: "TypeError: undefined is not an object (evaluating \'window. typeOfObject.data.overview\')"', 'source': '[string "..."]', 'js_error': "TypeError: undefined is not an object (evaluating 'window. typeOfObject.data.overview')", 'splash_method': 'evaljs', 'type': 'JS_ERROR', 'js_error_type': 'TypeError', 'js_error_message': "undefined is not an object (evaluating 'window. typeOfObject.data.overview')"}, 'description': 'Error happened while executing Lua script', 'error': 400, 'type': 'ScriptError'}```

**Comment by GaganBhat** [comment]
Did you ever find a solution @laranicolas ? Having the same exact issue here.
